Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6431,7 +6431,7 @@ void HTMLMediaElement::updatePlayState()
         return;
     }
 
-    bool shouldBePlaying = potentiallyPlaying();
+    bool shouldBePlaying = !m_paused && !m_pausedInternal && potentiallyPlaying();
```

Comment:
`m_pausedInternal` is always false here, considering the test above.

Reply:
m_paused can only be true if pauseInternal has been called, and is set in `HTMLMediaElement::pauseInternal()`
```
    if (!m_paused && !m_pausedInternal) {
        setPaused(true);
```

so to have m_paused to true, m_pausedInternal must be true as well, and we would return just above.

The other case where `HTMLMediaElement::setPaused` is called is when we currentTime has reached the end of the element. in `HTMLMediaElement::mediaPlayerTimeChanged` 

As such, I don't see how we could ever hit the case you describe.

Reply:
I thought of writing a test, but it is hard to write a test when you have something flaky happening 20% of the time.

And about it happening, it does happen, at least in our port with this test https://people.igalia.com/xrcalvar/tests/1132/mse_seek_zeroRate.html .

Reply:
This should be done in `HTMLMediaElement::potentiallyPlaying`

For the test: even if it only happened 20% of the time, with the 1000s of times a test is being repeated the failure would always end up showing if the bug was still there.